### ISSUE TITLE
Mechanism to notify user if sending an email has failed

### ISSUE THREAD

**Issue by ebruchez** [orbeon/orbeon-forms#367]
From user: "a nice enhancement would be a message to the user if the email sending fails, or at least a confirmation that it has successfully sent. To silently ignore such a failure is quite worrying and could undermine user confidence if they do not know if a request has been successfully executed."

At the very least, if there is an error reported immediately by the email API, we must notify the user right away.

It seems that at least in some cases the user gets a message. But make sure it's there are no errors that can be ignored.

Of course, some email errors cannot be detected right away, due to the architecture of email in general. In which case, are there some options to deal with that kind of things?

**Comment by ebruchez** [comment]
+1 from me today, see #7830. The `email()` process action should return a failure in this case, but apparently it doesn't.

It might also be an issue specific to the license form. If so, the form configuration should be modified and this issue closed. Otherwise, we should fix the issue.

**Comment by avernet** [comment]
> It might also be an issue specific to the license form.

Yes, it was. We were missing a proper recovery step in the `email` process for `orbeon/register`, which I have now added.

Regarding issue #367, the intended use case is still unclear. The core problem may be that our default `email` button process also lacks a recovery step, leaving users without an error message when something goes wrong. If that is what was initially reported, adding a `recover(...)` step to that process should resolve the issue.

Specifically, in `properties-form-runner.xml`, for the `oxf.fr.detail.process.email.*.*` process, we could add a `recover error-message("service-error")`.

**Comment by avernet** [comment]
- [x] Introduce a new, more specific, `email-error` message.
